geoh5py/workspace/workspace.py

```diff
@@ -678,50 +574,168 @@ def find_type(
         return found_type if isinstance(found_type, type_class) else None
 
-    def all_groups(self) -> List["group.Group"]:
-        """Get all active Group entities registered in the workspace.
+    @property
+    def ga_version(self) -> str:
         """
-        weakref_utils.remove_none_referents(self._groups)
-        return [cast("group.Group", v()) for v in self._groups.values()]
+        :obj:`str` Version of Geoscience Analyst software
+        """
+        return self._ga_version
 
-    def find_group(self, group_uid: uuid.UUID) -> Optional["group.Group"]:
+    @ga_version.setter
+    def ga_version(self, value: str):
+        self._ga_version = value
+
+    def get_entity(self, name: Union[str, uuid.UUID]) -> List[Optional[Entity]]:
         """
-        Find an existing and active Group object
+        Retrieve an entity from one of its identifier, either by name or uuid
+
+        :param name: Object identifier, either name or uuid.
+
+        :return: object_list: List of entities with the same given name
         """
-        return weakref_utils.get_clean_ref(self._groups, group_uid)
+        if isinstance(name, uuid.UUID):
+            list_entity_uid = [name]
 
-    def all_objects(self) -> List["object_base.ObjectBase"]:
-        """Get all active Object entities registered in the workspace.
+        else:  # Extract all objects uuid with matching name
+            list_entity_uid = [
+                key for key, val in self.list_entities_name.items() if val == name
+            ]
+
+        entity_list: List[Optional[Entity]] = []
+        for uid in list_entity_uid:
+            entity_list.append(self.find_entity(uid))
+
+        return entity_list
+
+    @property
+    def h5file(self) -> str:
         """
-        weakref_utils.remove_none_referents(self._objects)
-        return [cast("object_base.ObjectBase", v()) for v in self._objects.values()]
+        :str: Target *geoh5* file name with path
+        """
+        return self._h5file
 
-    def find_object(self, object_uid: uuid.UUID) -> Optional["object_base.ObjectBase"]:
+    @property
+    def list_data_name(self) -> Dict[uuid.UUID, str]:
         """
-        Find an existing and active Object
+        :obj:`dict` of UUID keys and name values for all registered Data
         """
-        return weakref_utils.get_clean_ref(self._objects, object_uid)
+        data_name = {}
+        for key, val in self._data.items():
+            entity = val.__call__()
+            if entity is not None:
+                data_name[key] = entity.name
+        return data_name
 
-    def all_data(self) -> List["data.Data"]:
-        """Get all active Data entities registered in the workspace.
+    @property
+    def list_entities_name(self) -> Dict[uuid.UUID, str]:
         """
-        weakref_utils.remove_none_referents(self._data)
-        return [cast("data.Data", v()) for v in self._data.values()]
+        :obj:`dict` of UUID keys and name values for all registered Entities
```

Comment:
missing periods, obj reference and return:
```suggestion
        :return: :obj:`dict` of :obj:`uuid.UUID` keys and name values for all registered Entities.
```